src/main/java/com/leverx/learningmanagementsystem/multitenancy/connection/datasource/RoutingDataSource.java

```diff
@@ -0,0 +1,67 @@
+package com.leverx.learningmanagementsystem.multitenancy.connection.datasource;
+
+import com.leverx.learningmanagementsystem.core.security.context.RequestContext;
+import com.zaxxer.hikari.HikariDataSource;
+import jakarta.annotation.PostConstruct;
+import jakarta.annotation.PreDestroy;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.jdbc.datasource.lookup.AbstractRoutingDataSource;
+
+import javax.sql.DataSource;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Slf4j
+@Setter
+public class RoutingDataSource extends AbstractRoutingDataSource {
+
+    private final Map<Object, Object> targetDataSources = new ConcurrentHashMap<>();
```

Comment:
it's better to define key as `String` and value as `DataSource` (because you are using only these types in your map)